spec/ruby-push-notifications/wns/wns_pusher_spec.rb

```diff
@@ -0,0 +1,58 @@
+module RubyPushNotifications
+  module WNS
+    describe WNSPusher do
+      let(:access_token) { 'token' }
+      let(:pusher) { WNSPusher.new(access_token) }
+
+      describe '#push' do
+        let(:device_urls) { [generate(:wns_device_url)] }
+        let(:notifications) { build_list(:wns_notification, 2, device_urls: ['http://hk2.notify.windows.com/1'], data: { message: { value1: 'hello' } }).to_a }
+        let(:raw_notification) { build :wns_notification }
+        let(:toast_data) { { title: 'Title', message: 'Hello WNS World!', type: :toast } }
+        let(:toast_notification) { build :wns_notification, device_urls: device_urls, data: toast_data }
+        let(:tile_data) { { message: 'Hello WNS World!', image: 'http://image.com/image.jpg', type: :tile } }
+        let(:tile_notification) { build :wns_notification, device_urls: device_urls, data: tile_data }
+        let(:badge_data) { { value: 10, type: :badge } }
+        let(:badge_notification) { build :wns_notification, device_urls: device_urls, data: badge_data }
+
+        let(:headers) {
+          {
+            'x-wns-notificationstatus' => 'Received',
+            'x-wns-deviceconnectionstatus' => 'Connected',
+            'x-wns-status' => 'Received'
+          }
+        }
+        let(:wns_connection) { RubyPushNotifications::WNS::WNSConnection }
+        let(:response) { [ { device_url: 'http://hk2.notify.windows.com/1', headers: headers, code: 200 } ]}
+
+        before do
+          stub_request(:post, %r{hk2.notify.windows.com}).
+            to_return status: [200, 'OK'], headers: headers
+        end
+
+        it 'submits every notification' do
+          pusher.push [raw_notification, toast_notification, tile_notification, badge_notification]
+          wns_connection::WP_TARGETS.keys.each do |push_type|
```

Comment:
Can we iterate here over the actual notifications being pushed one line above? This one is less robust I think as if someone empties the contents of `WP_TARGETS`, the test will always pass regardless. Does it make sense?

Reply:
I'm setted push types in spec manualy. Now if anyone remove types from WP_TARGETS - it wont' be passed